Challenge1-Vite/src/components/Filters.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+
+interface FiltersProps {
+  searchTerm: string;
+  setSearchTerm: (term: string) => void;
+  setSortOption: (option: string) => void;
+  setFilterFavorites: (isFavorite: boolean) => void;
+}
+
+const Filters: React.FC<FiltersProps> = ({ searchTerm, setSearchTerm, setSortOption, setFilterFavorites }) => {
+  const [isFavorites, setIsFavorites] = useState(false);
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(e.target.value);
+  };
+
+  const handleSortChange = (sortOption: string) => {
+    setSortOption(sortOption);
+    setDropdownOpen(false);
+  };
+
+  const handleFavoritesToggle = () => {
+    setIsFavorites(!isFavorites);
+    setFilterFavorites(!isFavorites);
```

Comment:
I'd suggest always grabbing the previous value from the parameter of the setter function. This can save us some trouble in the future

```suggestion
    setIsFavorites((prev) => !prev);
    setFilterFavorites((prev) => !prev);
```